subprojects/core/src/main/java/org/gradle/cache/internal/WrapperDistributionCleanupAction.java

```diff
@@ -81,21 +82,26 @@ public WrapperDistributionCleanupAction(File gradleUserHomeDirectory, UsedGradle
     }
 
     public void execute() {
+        long maximumTimestamp = Math.max(0, System.currentTimeMillis() - TimeUnit.DAYS.toMillis(1));
         Set<GradleVersion> usedVersions = this.usedGradleVersions.getUsedGradleVersions();
         Multimap<GradleVersion, File> checksumDirsByVersion = determineChecksumDirsByVersion();
         for (GradleVersion version : checksumDirsByVersion.keySet()) {
             if (!usedVersions.contains(version) && version.compareTo(GradleVersion.current()) < 0) {
-                deleteDistributions(checksumDirsByVersion.get(version));
+                deleteDistributions(checksumDirsByVersion.get(version), maximumTimestamp);
             }
         }
     }
 
-    private void deleteDistributions(Collection<File> dirs) {
+    private void deleteDistributions(Collection<File> dirs, long maximumTimestamp) {
         Set<File> parentsOfDeletedDistributions = Sets.newLinkedHashSet();
         for (File checksumDir : dirs) {
-            LOGGER.debug("Deleting distribution at {}", checksumDir);
-            if (FileUtils.deleteQuietly(checksumDir)) {
-                parentsOfDeletedDistributions.add(checksumDir.getParentFile());
+            if (checksumDir.lastModified() > maximumTimestamp) {
```

Comment:
While we could detect a download that's in progress, the failure could still occur after it has been completely downloaded but not yet started. Moreover, it would depend on wrapper internals so I think this safeguard is simple should be  ok.